File: mobatech-backend/models/medicine_category.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type MedicineCategory struct {
	ID          uint           `json:"id" gorm:"primarykey"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"deleted_at" gorm:"index"`
	Name        string         `json:"name"`
	Description string         `json:"description"`
	Icon        string         `json:"icon"`
}

File: mobatech-flutter/lib/features/pharmacy/models/medicine_category.dart
class MedicineCategory {
  final int id;
  final String name;
  final String icon;

  MedicineCategory({required this.id, required this.name, required this.icon});

  factory MedicineCategory.fromJson(Map<String, dynamic> json) {
    return MedicineCategory(
      id: json['id'] as int,
      name: json['name'] as String,
      icon: json['icon'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'name': name, 'icon': icon};
  }
}
